openmetadata-ui/src/main/resources/ui/src/hooks/paging/usePaging.ts

```diff
@@ -17,49 +17,104 @@ import {
   useMemo,
   useState,
 } from 'react';
+import { useHistory } from 'react-router-dom';
 import {
   INITIAL_PAGING_VALUE,
   PAGE_SIZE_BASE,
   pagingObject,
 } from '../../constants/constants';
+import { CursorType } from '../../enums/pagination.enum';
 import { Paging } from '../../generated/type/paging';
+import useCustomLocation from '../useCustomLocation/useCustomLocation';
+
+interface CursorState {
+  cursorType: CursorType | null;
+  cursorValue: string | null;
+}
+
+interface PagingHistoryStateData {
+  cursorData: CursorState;
+  currentPage: number | null;
+  pageSize: number | null;
+}
 
 export interface UsePagingInterface {
   paging: Paging;
   handlePagingChange: Dispatch<SetStateAction<Paging>>;
   currentPage: number;
-  handlePageChange: Dispatch<SetStateAction<number>>;
+  handlePageChange: (
+    page: number | ((page: number) => number),
+    cursorData?: CursorState
+  ) => void;
   pageSize: number;
   handlePageSizeChange: (page: number) => void;
   showPagination: boolean;
+  pagingCursor: PagingHistoryStateData;
 }
-
 export const usePaging = (
   defaultPageSize = PAGE_SIZE_BASE
 ): UsePagingInterface => {
+  // Extract initial values from history state if present
+  const history = useHistory();
+  const location = useCustomLocation();
+  const historyState = location.state as any;
+  const initialPageSize = historyState?.pageSize ?? defaultPageSize;
+  const initialCurrentPage = historyState?.cursorData?.cursorType
+    ? historyState.currentPage
+    : INITIAL_PAGING_VALUE;
+
   const [paging, setPaging] = useState<Paging>(pagingObject);
-  const [currentPage, setCurrentPage] = useState<number>(INITIAL_PAGING_VALUE);
-  const [pageSize, setPageSize] = useState(defaultPageSize);
+  const [currentPage, setCurrentPage] = useState<number>(initialCurrentPage);
+  const [pageSize, setPageSize] = useState<number>(initialPageSize);
+
+  const pagingCursorHistoryState: PagingHistoryStateData = {
+    cursorData: historyState?.cursorData,
+    currentPage: historyState?.currentPage,
+    pageSize: historyState?.pageSize,
+  };
 
   const handlePageSize = useCallback(
     (page: number) => {
       setPageSize(page);
       setCurrentPage(INITIAL_PAGING_VALUE);
+
+      // Update location state to persist pageSize for navigation
+      history.replace({
+        state: {
+          pageSize: page,
+        },
+      });
```

Comment:
On page size change, we are just setting the value of `pageSize` in the history state. Did you check if that will affect the current state values set for `cursorData` and `currentPage`?
Also, did we add a test case to check the state persists only if the page size is changed